index.js

```diff
@@ -40,29 +43,43 @@ client.on('ready', () => {
 // interaction has type: ChatInputCommandInteraction
 // https://old.discordjs.dev/#/docs/discord.js/main/class/ChatInputCommandInteraction
 client.on('interactionCreate', async interaction => {
-  if (!interaction.isChatInputCommand()) return;
-
   // defines the action to take when user enters /ping
-  console.log(interaction);
+  if (!interaction.isChatInputCommand()) return;
 
-  if (interaction.commandName === 'ping') {
-    await ping.execute(interaction);
-  }
+  const command = interaction.client.commands.get(interaction.commandName);
 
-  if (interaction.commandName === 'echo') {
-    await echo.execute(interaction);
+  if (!command) {
+    console.error(`No command matching ${interaction.commandName} was found.`);
+    return;
   }
 
-  if (interaction.commandName === 'queue') {
-    await queue.execute(interaction);
+  try {
+    await command.execute(interaction);
+  } catch (error) {
+    console.error(error);
+    await (interaction.replied || interaction.deferred
+      ? interaction.followUp({
+          content: 'There was an error while executing this command!',
+          ephemeral: true,
+        })
+      : interaction.reply({
+          content: 'There was an error while executing this command!',
+          ephemeral: true,
+        }));
   }
 });
 
 async function main() {
   try {
     console.log('Starting (/) commands refresh');
     await rest.put(Routes.applicationCommands(CLIENT_ID, GUILD_ID), {
-      body: commands,
+      body: all_commands.map(command => {
+        return {
+          name: command.data.name,
+          description: command.data.description,
+          options: command.data.options,
+        };
+      }),
     });
     client.login(TOKEN);
   } catch (error) {
```

Comment:
We can retry connection a couple times, and if doesn't work then we can try exiting instead of just logging the error.